pkg/plugin/system/system.go

```diff
@@ -41,23 +38,22 @@ type PluginSystem interface {
 }
 
 type pluginSystem struct {
-	finder        Finder
-	parser        Parser
-	clientFactory ClientFactory
-	clients       []ClientProvider
-	plugins       *PluginList
-	promptRunner  ioutils.PromptRunner
+	finder       loader.Finder
+	parser       loader.Parser
+	factory      client.Factory
```

Comment:
I prefer having it more explicit. Conceptually, we could have more factories for other things.

```suggestion
	clientFactory      client.Factory
```